src/test/java/org/eolang/speco/SpecoTest.java

```diff
@@ -0,0 +1,178 @@
+     *  but it is necessary to process all.
+     * @param pack Pack this test data
+     * @param temp Temporary test dir
+     * @throws IOException Iff IO error
+     */
+    @Disabled
+    @Tag("fast")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void convertsFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected transformation result",
+            Files.readString(SpecoTest.runSpeco(script, temp).resolve("app.eo")),
+            Matchers.equalTo(
+                script.get("after").toString()
+            )
+        );
+    }
+
+    @Tag("slow")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void compilesFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected execution result",
+            SpecoTest.exec(SpecoTest.runSpeco(script, temp).toString()),
+            Matchers.equalTo(
+                script.get("result").toString().split("\\r?\\n")
+            )
+        );
+    }
+
+    /**
+     * Runs Speco.
+     *
+     * @param script Yaml data object
+     * @param temp Path to the temporary dir
+     * @return Path to the output dir
+     * @throws IOException Iff IO error
+     */
+    private static Path runSpeco(final Map<String, Object> script, final Path temp)
+        throws IOException {
+        final Path input = temp.resolve("input");
+        final Path output = temp.resolve("output");
+        Files.createDirectories(input);
+        Files.write(
+            input.resolve("app.eo"),
+            script.get("before").toString().getBytes(),
+            StandardOpenOption.CREATE
+        );
+        new Speco(input, output, true).exec();
+        return output;
+    }
+
+    /**
+    * Compiles EO program.
+    *
+    * @param target Path to the dir with target EO program
+    * @return List of lines in output
+    * @throws IOException Iff IO error
+    */
+    private static String[] exec(final String target) throws IOException {
+        final String executor;
+        final String flag;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            executor = "cmd";
+            flag = "/c";
+        } else {
+            executor = "bash";
+            flag = "-c";
+        }
+        final Process process = new ProcessBuilder(
+            executor,
+            flag,
+            String.format("eoc link -s %s && eoc --alone dataize app && eoc clean", target)
+        ).start();
+        final StringWriter writer = new StringWriter();
+        IOUtils.copy(process.getInputStream(), writer);
```

Comment:
@advasileva Since you are using `InputStream` here, I highly recommend to close it and check if you leave any other open files after tests, it could be a cause of problem with JUnit temporary folders.

Reply:
@volodya-lombrozo Closing the streams in SpecoTest did not help, I will try to close the files

Reply:
@advasileva I also highly recommend to use [Process.waitFor](https://docs.oracle.com/javase/8/docs/api/java/lang/Process.html#waitFor--) method.

Reply:
@advasileva As I see from logs we have that exception most of the time:
```
SpecoTest.compilesFromEo(String, Path)[3] � IO Failed to delete temp directory C:\Users\RUNNER~1\AppData\Local\Temp\junit18001172281059271577. The following paths could not be deleted (see suppressed exceptions for details): , input_prs_aoi
```

This files are created by that method:
```java
 private static Path parse(final Path input) throws IOException {
        final StringBuilder name = new StringBuilder(input.toString());
        final Path source = Path.of(name.append("_prs").toString());
        FileUtils.copyDirectory(input.toFile(), source.toFile());
        for (final Path path : Files.newDirectoryStream(source)) {
            new Syntax(
                "scenario",
                new InputOf(String.format("%s\n", Files.readString(path))),
                new OutputTo(new FileOutputStream(path.toFile()))
            ).parse();
        }
        LauncherKt.launch(source.toString());
        return Path.of(name.append("_aoi").toString());
    }
```

Maybe we have also close  `Files.newDirectoryStream(source)` stream. It's just a guess.

Reply:
I [debug](https://github.com/advasileva/speco/pull/6) this problem, it turned out that `LauncherKt.launch(source.toString());` is a process that does not allow deleting files. I do not know how to fix this from java (`FORCE` removal did not help), so I suggest refuse to use `@TempDir` or set a `cleanup = CleanupMode.NEVER` policy

Reply:
@volodya-lombrozo check out the message above